cmd/virt-prerunner/main.go

```diff
@@ -279,6 +304,37 @@ func buildVMConfig(ctx context.Context, vm *virtv1alpha1.VirtualMachine) (*cloud
 						vmConfig.Devices = append(vmConfig.Devices, &sriovDeviceConfig)
 					}
 				}
+			case iface.Vhostuser != nil:
+				socket := os.Getenv("VHOST_USER_SOCKET")
+				if socket == "" {
+					return nil, fmt.Errorf("vhostuser socket path not found")
+				}
+				netConfig := cloudhypervisor.NetConfig{
+					Id:          iface.Name,
+					Mac:         iface.MAC,
+					VhostUser:   true,
+					VhostMode:   "server",
+					VhostSocket: socket,
+				}
+				vmConfig.Net = append(vmConfig.Net, &netConfig)
+
+				memNodeSet, err := cpuset.GetMEMNodeSet()
+				if err != nil {
+					return nil, fmt.Errorf("get memory node set: %s", err)
+				}
+				nodes := memNodeSet.ToSlice()
+				if len(nodes) != 1 {
+					return nil, fmt.Errorf("may not use more than 1 numa node")
+				}
+				vmConfig.Memory.Zones = []*cloudhypervisor.MemoryZoneConfig{{
```

Comment:
Is it necessary to set memory zone if the VM's memory would be controlled by cgroup of the container?

Reply:
Thanks for reminding! The flag `memory-zone` should be removed.